server/src/main/java/org/elasticsearch/action/search/SearchResponseMerger.java

```diff
@@ -110,13 +110,20 @@ int numResponses() {
     }
 
     /**
-     * Returns the merged response. To be called once all responses have been added through {@link #add(SearchResponse)}
-     * so that all responses are merged into a single one.
+     * Returns the merged response of all SearchResponses received so far. This can be called at any point,
+     * including when only some clusters have finished, in order to get "incremental" partial results.
      */
-    SearchResponse getMergedResponse(Clusters clusters) {
+    public SearchResponse getMergedResponse(Clusters clusters) {
+        return getMergedResponse(clusters, null);
+    }
+
+    /**
+     * TODO: DOCUMENT ME
+     */
+    public SearchResponse getMergedResponse(Clusters clusters, SearchResponse additionalResponse) {
```

Comment:
I feel like this new method complicates a bit the design of `SearchResponseMerger`. What are the downsides of creating a new one providing it all the responses + the artifically built one with aggs? As opposed to the existing one, it should be a short lived object? Perhaps we could expose that as a static method?

Reply:
That would work, but seems less desirable to me. The core goal of the new refactoring was to share a single SearchResponseMerger. In addition, this new idea would more-or-less break encapsulation as we now have to pass out all of the inner state of the SearchResponseMerger to another one. I guess I could do it as a copy constructor.